### PR TITLE
Allow Julia to build on LLVM 13

### PR DESCRIPTION
**Pull request by vchuravy**
Adapts to changes and also updates most of the patch list.

### CODE REVIEW

**Review by vtjnash — APPROVED** [review]

File: deps/llvm.mk [hunk 1]
```diff
@@ -508,7 +508,7 @@ $(eval $(call LLVM_PATCH,llvm-D88630-clang-cmake))
 endif
 $(eval $(call LLVM_PATCH,llvm-invalid-addrspacecast-sink)) # Still being upstreamed as D92210
 $(eval $(call LLVM_PATCH,llvm-11-D93154-globalisel-as))
-$(eval $(call LLVM_PATCH,llvm-11-D94813-mergeicmps))
+$(eval $(call LLVM_PATCH,llvm-11-D94813-mergeicmps)) # remove for LLVM 14
 $(eval $(call LLVM_PROJ_PATCH,llvm-11-AArch64-FastIsel-bug))
 $(eval $(call LLVM_PATCH,llvm-12-D97435-AArch64-movaddrreg))
 $(eval $(call LLVM_PROJ_PATCH,llvm-11-D97571-AArch64-loh)) # remove for LLVM 13
```
Comment (Seelengrab) [inline deps/llvm.mk:514]: > # remove for LLVM 13

Missed this?

File: deps/llvm.mk [hunk 2]
```diff
@@ -517,6 +517,20 @@ $(eval $(call LLVM_PROJ_PATCH,llvm-12-fde-symbols-aarch64)) # remove for LLVM 13
 $(eval $(call LLVM_PROJ_PATCH,llvm-12-force-eh_frame-aarch64)) # remove for LLVM 13
```
Comment (Seelengrab) [inline deps/llvm.mk:517]: And this?

File: deps/llvm.mk (hunk 1 above)
Comment (vchuravy) [reply deps/llvm.mk:514]: You are looking at the patch list for LLVM 12

File: deps/llvm.mk [hunk 3]
```diff
@@ -517,6 +517,20 @@ $(eval $(call LLVM_PROJ_PATCH,llvm-12-fde-symbols-aarch64)) # remove for LLVM 13
...
 endif # LLVM_VER 12.0
 
+ifeq ($(LLVM_VER_SHORT),13.0)
+$(eval $(call LLVM_PATCH,llvm7-revert-D44485)) # Needs upstreaming
+$(eval $(call LLVM_PROJ_PATCH,llvm-13-D75072-SCEV-add-type))
+$(eval $(call LLVM_PATCH,llvm-julia-tsan-custom-as))
```
Comment (vchuravy) [inline deps/llvm.mk:523]: Do we want to drop this for now, and instead revert the ptrtoint/inttoptr change so that we get errors if this is needed?

Comment (vchuravy) [reply deps/llvm.mk:523]: Fat fingered on my phone. I meant the patch one line higher.

Comment (vtjnash) [reply deps/llvm.mk:523]: it is likely still needed (and there is a likely LLVM reviewer now), it just needs to be generalized to all pointers, not just NI

**Review by maleadt — APPROVED** [review]

**Pull request merged**